pkg/thanos/operator.go

```diff
@@ -803,3 +806,27 @@ func (o *Operator) enqueueForNamespace(store cache.Store, nsName string) {
 		)
 	}
 }
+
+func ApplyConfigurationFromThanosRuler(a *monitoringv1.ThanosRuler) *monitoringv1ac.ThanosRulerApplyConfiguration {
+	trac := monitoringv1ac.ThanosRulerStatus().
+		WithPaused(a.Status.Paused).
+		WithPaused(a.Status.Paused).
+		WithReplicas(a.Status.Replicas).
+		WithAvailableReplicas(a.Status.AvailableReplicas).
+		WithUpdatedReplicas(a.Status.UpdatedReplicas).
+		WithUnavailableReplicas(a.Status.UnavailableReplicas)
+
+	for _, condition := range a.Status.Conditions {
+		trac.WithConditions(
```

Comment:
(suggestion) this can be extracted to pkg/operator/operator.go (or a new  pkg/operator/applyconfiguration.go) and shared with Alertmanager.